system/alternatives.py

```diff
@@ -114,7 +124,16 @@ def main():
                         link = line.split()[1]
                         break
 
-    if current_path != path:
+    if not force:
+        need_to_force = False
+    elif not os.path.islink(link):
```

Comment:
well arguably we could always provide the --force flag, although it makes no difference. the point of this logic was more to do with detecting if a change was required, which requires extra checks (the islink and readlink calls here), but only if --force is specified.